src/decode/ykpt/mod.rs

```diff
@@ -0,0 +1,61 @@
+//! The Yk PT trace decoder.
+
+use crate::{decode::TraceDecoder, errors::HWTracerError, Block, Trace};
+
+mod packet_parser;
+use packet_parser::PacketParser;
+
+pub(crate) struct YkPTTraceDecoder {}
+
+impl TraceDecoder for YkPTTraceDecoder {
+    fn new() -> Self {
+        Self {}
+    }
+
+    fn iter_blocks<'t>(
+        &'t self,
+        trace: &'t dyn Trace,
+    ) -> Box<dyn Iterator<Item = Result<Block, HWTracerError>> + '_> {
+        let itr = YkPTBlockIterator {
+            errored: false,
+            parser: PacketParser::new(trace.bytes()),
+        };
+        Box::new(itr)
+    }
+}
+
+/// Iterate over the blocks of an Intel PT trace using the fast Yk PT decoder.
+struct YkPTBlockIterator<'t> {
+    /// Error state. Set to true when an error occurs, thus invalidating the iterator.
```

Comment:
Could just be `/// Set to true when an error has occurred`?

Reply:
Updating it in both places in this PR is fine.

Reply:
New code fix in [3567abf](https://github.com/ykjit/hwtracer/pull/105/commits/3567abfb722b6106c105a07bc9a0b4ceda47d3a4)

Existing code fix in [09ce652](https://github.com/ykjit/hwtracer/pull/105/commits/09ce652c772e616b5931dcebf31c359e5de2e1c6)